extensions/dedup_messages.go

```diff
@@ -0,0 +1,123 @@
+package extensions
+
+import (
+	"context"
+	"crypto/tls"
+	"fmt"
+	"hash/fnv"
+	"strconv"
+	"time"
+
+	"github.com/redis/go-redis/v9"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/viper"
+	"github.com/topfreegames/pusher/interfaces"
+)
+
+// Dedup struct
+type dedup struct {
+	redis             *redis.Client
+	ttl               time.Duration
+	statsReporters    []interfaces.StatsReporter
+	l                 *logrus.Entry
+	gamePercentages   map[string]int
+}
+
+func NewDedup(ttl time.Duration, config *viper.Viper, statsReporters []interfaces.StatsReporter, logger *logrus.Logger) dedup {
+	host := config.GetString("dedup.redis.host")
+	port := config.GetInt("dedup.redis.port")
+	pwd := config.GetString("dedup.redis.password")
+	disableTLS := config.GetBool("dedup.tls.disabled")
+
+	addr := fmt.Sprintf("%s:%d", host, port)
+	opts := &redis.Options{
+		Addr:     addr,
+		Password: pwd,
+		DB:       1,
+	}
+
+	if !disableTLS {
+		opts.TLSConfig = &tls.Config{}
+	}
+
+	gamePercentages := make(map[string]int)
+	games := config.GetStringMap("dedup.games")
+	for game := range games {
+		if config.IsSet("dedup.games." + game + ".percentage") {
+			gamePercentages[game] = config.GetInt("dedup.games." + game + ".percentage")
+		} else {
```

Comment:
just do `config.SetDefault` to 0 and read with `GetInt`. Or, read directly with `GetInt` and treat if within boundaries `p > 0 && p < 100`, if someone set -1 you'd discard or just return `true` in `IsUnique` (fail close)